test/e2e/serial/tests/configuration.go

```diff
@@ -343,5 +345,137 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
+			Expect(err).ToNot(HaveOccurred())
+
+			targetedKCs := make([]*machineconfigv1.KubeletConfig, 0)
+			for _, mcp := range mcps {
+				for i := 0; i < len(kcList.Items); i++ {
+					kc := &kcList.Items[i]
+					kcMcpSel, err := metav1.LabelSelectorAsSelector(kc.Spec.MachineConfigPoolSelector)
+					Expect(err).ToNot(HaveOccurred())
+
+					if kcMcpSel.Matches(labels.Set(mcp.Labels)) {
+						targetedKCs = append(targetedKCs, kc)
+					}
+				}
+			}
+			Expect(len(targetedKCs)).ToNot(BeZero(), "there should be at least one kubeletconfig.machineconfiguration object")
+
+			By("modifying reserved CPUs under kubeletconfig")
+			initialRsvCPUs := make(map[string]string)
+			for i := 0; i < len(targetedKCs); i++ {
+				kcAsMc := targetedKCs[i]
+				kcObj, err := kubeletconfig.MCOKubeletConfToKubeletConf(kcAsMc)
+				Expect(err).ToNot(HaveOccurred())
+
+				initialRsvCPUs[kcAsMc.Name] = kcObj.ReservedSystemCPUs
+				applyNewReservedSystemCPUsValue(&kcObj.ReservedSystemCPUs)
+				err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, kcAsMc)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = fxt.Client.Update(context.TODO(), kcAsMc)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("waiting for MachineConfigPools to get updated")
+			var wg sync.WaitGroup
+			for _, mcp := range mcps {
+				wg.Add(1)
+				go func(mcpool *machineconfigv1.MachineConfigPool) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					err = e2ewait.ForMachineConfigPoolCondition(fxt.Client, mcpool, machineconfigv1.MachineConfigPoolUpdated, configuration.MachineConfigPoolUpdateInterval, configuration.MachineConfigPoolUpdateTimeout)
+					Expect(err).ToNot(HaveOccurred())
+				}(mcp)
+			}
+			wg.Wait()
+
+			By("checking that NodeResourceTopology objects available resources has changed")
+			for _, initialNrt := range initialNrtList.Items {
+				wg.Add(1)
+				go func(nrt *nrtv1alpha1.NodeResourceTopology) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					Eventually(func() (bool, error) {
+						updatedNrt := &nrtv1alpha1.NodeResourceTopology{}
+						err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(nrt), updatedNrt)
+						Expect(err).ToNot(HaveOccurred())
+						// since the tests are not running in parallel, we know that nothing can pollute
+						// the current NRT object, and therefore it is guaranteed that the difference between
+						// the initial and updated NRT object derived from the change in the reserved CPUs under the
+						// kubeletconfig. we don't care from the exact accounting, we just want to make sure that there is
+						// a difference.
+						equal, err := e2enrt.CheckEqualAvailableResources(*nrt, *updatedNrt)
+						if err != nil {
+							klog.Warningf("failed to check equal available resources of NodeResourceTopology %q", updatedNrt.Name)
+							return false, err
+						}
+						if equal == true {
+							klog.Warningf("NodeResourceTopology %q has not been updated after kubelet modification", updatedNrt.Name)
```

Comment:
This works but seems a bit too fragile. I wonder if it is more explicit/safer to check the `configmap`.
Or, we can add a new `status` field in the operator to report which version of the `configmap` the operator is using.
Let's think about options.

Reply:
> we can add a new status field in the operator to report which version of the configmap the operator is using.
Let's think about options.

That sounds like a good suggestion, but there are some open questions:
1. How do we track the version number? 
2. What is the version convention?
3. What happens if the operator gets restarted? or if the configmap gets restarted? (without changes)

Reply:
it could be sufficient to expose the hash signature of the configmap the operator is using. This acts as the simplest validation mechanism, and sidesteps all the (real) issues outlined above.

Reply:
Actually, it is much easier to look for the new ```SystemCPUsReserved``` value in the ```configmap``` 
But adding the configmap hash as one of the status is a good idea in general

Reply:
true, but looking into that value in the `configmap` will tell us what the operator is _supposed_ to use, not what is actually using

Reply:
The same applies for checking the ```configmap``` hash under the ```status```

Reply:
how? `status` report what the operator (/controller) is using, so it reflects somehow its internal state. OTOH we can also dump the content of the configmap in the status, but if we go this way I'd just start with the signature

Reply:
The ```configmap``` gets **created** by the operator, then a fortiori is being used by it :)

Reply:
> but if we go this way I'd just start with the signature

In progress

Reply:
right, got sidetracked into a tunnel vision. Will re-evaluate this whole subthread. Feel free to pause this work.